proposals/zips/zip007.sol

```diff
@@ -0,0 +1,122 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import "@forge-std/console.sol";
+
+import {Addresses} from "@proposals/Addresses.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+import {TimelockProposal} from "@proposals/proposalTypes/TimelockProposal.sol";
+import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.sol";
+
+contract zip007 is Proposal, TimelockProposal {
+    string public name = "ZIP007";
+    string public description = "ZTX CGv1.2.4 MaxSupply updates";
+
+    struct TokenIDMaxSupplySettings {
+        uint256 maxSupply;
+        uint256 tokenId;
+    }
+
+    TokenIDMaxSupplySettings[] private wearableTokenIDMaxSupplySettings;
+
+    function setAndConfirmWearableData() private {
+        // Wearable data
+        string memory wearableData = string(
+            abi.encodePacked(vm.readFile("./proposals/zips/zip007.json"))
+        );
+
+        bytes memory parsedJson = vm.parseJson(wearableData);
+
+        TokenIDMaxSupplySettings[] memory wearablesDecoded = abi.decode(
+            parsedJson,
+            (TokenIDMaxSupplySettings[])
+        );
+
+        for (uint256 i = 0; i < wearablesDecoded.length; i++) {
+            wearableTokenIDMaxSupplySettings.push(
+                TokenIDMaxSupplySettings(
+                    wearablesDecoded[i].maxSupply,
+                    wearablesDecoded[i].tokenId
+                )
+            );
+        }
+
+        // sanity checks
+        assertEq(wearableTokenIDMaxSupplySettings.length, 1, "Invalid wearableTokenIDMaxSupplySettings length");
+
+        uint maxSupplyTotal = 0;
+
+        // sum numbers from requrements sheet
+        for (uint256 i = 0; i < wearableTokenIDMaxSupplySettings.length; i++) {
+            maxSupplyTotal += wearableTokenIDMaxSupplySettings[i].maxSupply;
+        }
+
+        assertEq(maxSupplyTotal, 75, "Invalid maxSupplyTotal");
+    }
+
```

Comment:
Should think about turning this into a library, so that we're not repeating ourselves for every  wearable/placeable/consumable max supply update.